lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +32,134 @@ const _changesRequiredOutput = """
 Then, review the the changes, address any FIXMEs, and commit.
 """;
 
+const String react16CodemodName = 'React16 Codemod';
+
+YamlMap pubspecYaml;
+
+Map<String, List<VersionConstraint>> packagesToCheckFor = {
+  'react': [
+    VersionConstraint.parse('^4.0.0'),
+    VersionConstraint.parse('^5.0.0')
+  ],
+  'over_react': [
+    VersionConstraint.parse('^2.0.0'),
+    VersionConstraint.parse('^3.0.0')
+  ]
+};
+
+VersionConstraint getDependencyVersion(String packageName) {
+  if (pubspecYaml == null) return null;
+
+  try {
+    if (pubspecYaml['dependencies'] != null &&
+        pubspecYaml['dependencies'][packageName] != null) {
+      return VersionConstraint.parse(pubspecYaml['dependencies'][packageName]);
+    } else if (pubspecYaml['dev_dependencies'] != null &&
+        pubspecYaml['dev_dependencies'][packageName] != null) {
+      return VersionConstraint.parse(
+          pubspecYaml['dev_dependencies'][packageName]);
+    }
+  } catch (e, stackTrace) {
+    react16CodemodLogger.warning('Unexpected Error.', e, stackTrace);
+  }
+  return null;
+}
+
 void main(List<String> args) {
-  final query = FileQuery.dir(
-    pathFilter: isDartFile,
-    recursive: true,
-  );
-  exitCode = runInteractiveCodemodSequence(
-    query,
-    [
-      ReactDomRenderMigrator(),
-      ReactStyleMapsUpdater(),
-    ],
-    args: args,
-    defaultYes: true,
-    changesRequiredOutput: _changesRequiredOutput,
-  );
-
-  final logger = Logger('over_react_codemod.fixmes');
-  for (var dartFile in query.generateFilePaths()) {
-    final dartSource = File(dartFile).readAsStringSync();
-    if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
-      logger.severe(
-          'over_react_codemod validation comments are unaddressed in $dartFile');
-      exitCode = 1;
+  Map<String, bool> inTransition = {};
+  bool foundReactOrOverReact = false;
+
+  react16CodemodLogger.onRecord.listen((rec) {
+    // These need to use `print`, not stdio, so that they are picked up properly
+    // by the `test` package's custom print handler.
+    // Otherwise, these logs interfere with parsing test's JSON output.
+    if (rec.message != '') {
+      final prefix = '[${rec.level}] ${rec.loggerName}: ';
+      print('$prefix${rec.message}');
+    }
+    if (rec.error != null) {
+      print(rec.error.toString());
+    }
+    if (rec.stackTrace != null) {
+      print(rec.stackTrace.toString());
```

Comment:
#nit toString is redundant when using print
```suggestion
      print(rec.stackTrace);
```